### PR TITLE
[WIP] checksum API: Generalize to openPMD

### PR DESCRIPTION
**Pull request by ax3l**
Add support for openPMD alongside the plotfile support to the checksum API.

Author: Maxence Thévenet (added with `git --author=...`) :sparkling_heart:

### CODE REVIEW

File: Regression/Checksum/checksum.py [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
- License: BSD-3-Clause-LBNL
- """
+This file is part of WarpX.
 
+License: BSD-3-Clause-LBNL
+"""
 from benchmark import Benchmark
-import yt
+from backend.amrex_backend import Backend
+import re
```
Comment (ax3l) [inline Regression/Checksum/checksum.py]: I think that's the unused import:
```suggestion
```

**Comment by ax3l** [comment]
@MaxThevenet it looks like this changes the checksum of `galilean_hybrid_2d`

**Comment by MaxThevenet** [comment]
Thanks for this PR! Could you also include the changes to files in `hipace/tests/checksum/` made in this PR? This will make the naming more consistent: remove mentions of `plotfiles` for backend-agnostic parts of the code, and maybe fix the Galilean test. Note that the Galilean test is handled in a special way within the checksum library, so it is not a surprise that something goes wrong for this test only.

**Comment by ax3l** [comment]
Thanks! I think I updated both files, did I miss one?

**Comment by MaxThevenet** [comment]
Hipace PR 291 also contains changes to `tests/checksum/checksum.py`, as well as to `tests/checksum/checksumAPI.py`. I think these should also be imported back here. Hopefully this will fix the bug, otherwise I'll have a look.

**Comment by ax3l** [comment]
Thanks! I think that was changed by another PR - yet it's exactly what was missing :tada: 
Thank you!

**Comment by ax3l** [comment]
Update now, looks like the `galilean_hybrid_2d` is the only one failing

**Comment by ax3l** [comment]
@MaxThevenet I probably need your help on this one. Do you have time to go through it with me at some point? :)

File: Regression/Checksum/backend/amrex_backend.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        '''
+
+        return self.dataset.max_level+1;
+
+    def get_field_checksum(self, lev, field, test_name):
+        ''' Calculate the checksum for a given field at a given level in the dataset
+        '''
+
+        # Boolean variable parsed from test name needed for workaround below
+        galilean = True if re.search('galilean', test_name) else False
```
Comment (MaxThevenet) [inline Regression/Checksum/backend/amrex_backend.py]: For debugging, could you print `test_name` here, as well as the value of `Galilean`? Maybe, for some reason, the `test_name` was changed when we refactored this checksum library.

Comment (MaxThevenet) [reply Regression/Checksum/backend/amrex_backend.py]: OK that didn't help, `galilean` has the right value.

File: Regression/Checksum/backend/amrex_backend.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                np.array([grid.LeftEdge.v for grid in lev_grids]), axis=0)
+            all_data_level = self.dataset.covering_grid(
+                level=lev, left_edge=LeftEdge, dims=self.dataset.domain_dimensions)
+            Q = all_data_level[field].v.squeeze()
+            return np.sum(np.abs(Q))
+        # Workaround for Galilean tests: the standard procedure above
+        # does not seem to read 2D fields data correctly
+        elif (galilean):
+            Q = self.dataset.index.grids[lev][field].v.squeeze()
+            return np.sum(np.abs(Q))
```
Comment (EZoni) [inline Regression/Checksum/backend/amrex_backend.py]: @ax3l I think I removed this workaround in #1536, about two months ago. I think we should not (re-)introduce a difference between Galilean and non-Galilean here. This might explain why something weird is happening with the test `galilean_hybrid_2d`. This is the only Galilean test with domain decomposition, which causes a problem with the reading of data as it is done currently here, which was fixed in #1536 indeed (see PR description for more info on why this implementation was wrong).

In other words, I think you might have moved this piece of code here (with the dichotomy `if not (galilean) ... elif (galilean)`) from a version of checksum.py that was older than #1536, where it was modified. Then, while rebasing this PR on development, it is possible that git missed the conflict because the same piece of code was moved here into a different file, so git would not catch that there was a conflict in the changes. Does this make sense to you?

Comment (MaxThevenet) [reply Regression/Checksum/backend/amrex_backend.py]: Ah very good catch, thanks @EZoni. The files in this PR come from Hipace++, and were copied from WarpX months ago, so I think you're totally right on what happened. Let's just include your changes from #1536 here, and then I'll also do them in Hipace++. The checksum module is much cleaner without this special Galilean handling indeed.

File: Regression/Checksum/checksum.py [hunk 4]
```diff
@@ -48,39 +44,27 @@ def read_plotfile(self, do_fields=True, do_particles=True):
         @param do_particles Whether to read particles from the plotfile.
         '''
 
-        ds = yt.load(self.plotfile)
-        grid_fields = [item for item in ds.field_list if item[0] == 'boxlib']
-        species_list = set([item[0] for item in ds.field_list if
-                            item[1][:9] == 'particle_' and item[0] != 'all'])
+        ds = Backend(self.plotfile)
+        grid_fields = ds.fields_list()
```
Comment (EZoni) [inline Regression/Checksum/checksum.py:48]: Looking at the lines that were removed, maybe the correct name was `ds.field_list`? (the error on CI in fact says that "name 'grid_fields' is not defined")
```suggestion
        grid_fields = ds.field_list()
```

**Comment by RemiLehe** [comment]
Superseded by https://github.com/ECP-WarpX/WarpX/pull/4519
